[FIX] l10n_it_fatturapa_in - Gruppi IVA

SirAionTech opened the pull request:
Risolve https://github.com/OCA/l10n-italy/issues/1667 per `16.0`.
Risolve https://github.com/OCA/l10n-italy/issues/3146 per `16.0`.

Sostituisce https://github.com/OCA/l10n-italy/pull/3513 per risolvere https://github.com/OCA/l10n-italy/pull/3513#pullrequestreview-1687749596.

Forward port di https://github.com/OCA/l10n-italy/pull/3100.
Ho dovuto cambiare qualcosa durante il porting, soprattutto nell'ultimo commit, perché dopo https://github.com/OCA/l10n-italy/pull/2750 non viene più sollevata un'eccezione quando vengono trovati partner duplicati.

Dipende da:
- [x] https://github.com/OCA/partner-contact/pull/1742

SirTakobi committed:
[FIX] l10n_it_fatturapa_in: Manage VAT groups

Co-authored-by: SirAionTech <simone.rubino@aion-tech.it>

SirTakobi committed:
[REF] l10n_it_fatturapa_in: Include partners in error message
Extract commercial partner retrieval
Extract partner search
Use the actual domain from multi-company partners sharing rule

Co-authored-by: SirAionTech <simone.rubino@aion-tech.it>

SirTakobi committed:
[IMP] l10n_it_fatturapa_in: Skip Duplicated Partners for some fields
Only CedentePrestatore and CessionarioCommittente are checked for VAT Groups,
when other partner fields match multiple partners just log an inconsistency

Co-authored-by: SirAionTech <simone.rubino@aion-tech.it>

SirTakobi committed:
[COV] l10n_it_fatturapa_in: Error when there are multiple partners

Co-authored-by: SirAionTech <simone.rubino@aion-tech.it>

SirAionTech commented:
I test falliscono per https://github.com/OCA/l10n-italy/issues/3801.

SirAionTech committed:
[COV] l10n_it_fatturapa_in: Do not depend on previous tests

francesco-ooops commented:
@SirAionTech qualche indicazione per test funzionali?

SirAionTech commented:
> @SirAionTech qualche indicazione per test funzionali?

Sono nelle issue linkate in descrizione

> Risolve #1667 per `16.0`.
Risolve #3146 per `16.0`.

SirAionTech commented:
/ocabot rebase

SirAionTech commented:
@matteoopenf @eLBati magari vi interessa visto che avete revisionato https://github.com/OCA/l10n-italy/pull/3100?

matteoopenf commented:
> @matteoopenf @eLBati magari vi interessa visto che avete revisionato #3100?

si mi interessa ci arrivo

patrickt-oforce commented:
@SirAionTech riesci a fare un rebase per riavviare runbot?

francesco-ooops commented:
/ocabot rebase

patrickt-oforce commented:
Test funzionale effettuato su runbot con file contenente un partner di un gruppo IVA correttamente importato e il partner viene correttamente riconosciuto

matteoopenf commented:
se entra vediamo per fare il back port a 14 e chiudiamo quella inizialmente aperta da noi

francesco-ooops commented:
@OCA/local-italy-maintainers buona per il merge?

TheMule71 commented on l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py:
```diff
@@ -213,109 +214,163 @@ def check_partner_base_data(self, partner_id, DatiAnagrafici):
+        if len(partners) > 1 and not commercial_partner:
+            found_partner = partner_model.browse()
+        elif commercial_partner:
+            commercial_partner_id = commercial_partner.id
```
`commercial_partner_id`  come variabile non serve, basterebbe mettere `commercial_partner.id` sotto

monen17 replied:
:+1: fatto

TheMule71 commented on l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py:
```diff
@@ -213,109 +214,163 @@ def check_partner_base_data(self, partner_id, DatiAnagrafici):
+            vals["name"] = DatiAnagrafici.Anagrafica.Denominazione
+        return vals
+
+    def getPartnerBase(self, DatiAnagrafici):  # noqa: C901
```
`# noqa: C901` probabilmente adesso dopo il refactoring non serve piu

monen17 replied:
:+1: fatto

matteoopenf commented:
@SirPyTech per caso riesci a fare i cambiamenti mi sembra che sia il tuo nuovo nick dopo @SirAionTech
Grazie Mille

SirPyTech commented:
> @SirPyTech per caso riesci a fare i cambiamenti mi sembra che sia il tuo nuovo nick dopo @SirAionTech Grazie Mille

Sì è il nuovo nick, però non è detto che io possa seguire i lavori fatti con i vecchi account.

Consiglio di trattare @SirAionTech come se fosse un account disattivato (di fatto lo è): se ci sono modifiche da fare qualcuno dovrà sostituire la PR, ovviamente rispettando l'authorship e quant'altro.
Può capitare che lo faccia io, ma come me possono farlo anche altri contributors.

matteoopenf commented:
> > @SirPyTech per caso riesci a fare i cambiamenti mi sembra che sia il tuo nuovo nick dopo @SirAionTech Grazie Mille
> 
> Sì è il nuovo nick, però non è detto che io possa seguire i lavori fatti con i vecchi account.
> 
> Consiglio di trattare @SirAionTech come se fosse un account disattivato (di fatto lo è): se ci sono modifiche da fare qualcuno dovrà sostituire la PR, ovviamente rispettando l'authorship e quant'altro. Può capitare che lo faccia io, ma come me possono farlo anche altri contributors.

ok

matteoopenf commented:
@TheMule71  sono mandatorie le modifiche che hai chiesto? o potremmo comunque mergiare?

monen17 reviewed (commented):
Grazie @TheMule71 della revisione, ho fatto le modifiche che suggeristi.
Secondo me comunque non erano bloccanti per il merge.
Puoi vedere se ora va meglio?

Ho anche risolto i conflitti.

monen17 commented:
I test non partono per https://github.com/OCA/l10n-italy/issues/4519#issuecomment-3083392333.

francesco-ooops commented:
/ocabot rebase

monen17 committed:
[IMP] l10n_it_fatturapa_in: Remove tests dependency

Before this change, the test finds two different partners with two
different commercial partners that are created by other tests so no
supplier is assigned to the created bill

francesco-ooops commented:
@OCA/local-italy-maintainers dato che c'è andato abbastanza lavoro, mergiamo?

monen17 commented:
/ocabot rebase

monen17 commented:
@OCA/local-italy-maintainers la questione dei gruppi IVA ogni tanto rispunta, questa PR ha fatto da poco 2 anni :candle::candle: e porta le modifiche che sono già state mergiate in `12.0`, si può portare avanti?

@TheMule71 avevi già dato un'occhiata (https://github.com/OCA/l10n-italy/pull/3810#pullrequestreview-2666647283 e https://github.com/OCA/l10n-italy/pull/3810#pullrequestreview-2666649454) e ho risposto quasi 6 mesi fa (https://github.com/OCA/l10n-italy/pull/3810#pullrequestreview-3149258515), puoi vedere se ora va bene?

Grazie!

patrickt-oforce commented:
@monen17 / @SirAionTech questa PR è stata fatta anche per 14.0?

SirPyTech commented:
> @monen17 / @SirAionTech questa PR è stata fatta anche per 14.0?

Dalle issues ho trovato la tua https://github.com/OCA/l10n-italy/pull/3513 che però si è chiusa per inattività

patrickt-oforce commented:
> > @monen17 / @SirAionTech questa PR è stata fatta anche per 14.0?
> 
> Dalle issues ho trovato la tua #3513 che però si è chiusa per inattività

Si si; non sapevo poi se era stata fatta una nuova PR. Tanto devo aprirne un'altra per un problema legato al multicompany, riparto dal modulo base, e riporto i 4 commit dalla #3100 integrando poi il check per il multicompany o converrebbe fare il backport di questa? Ma vedo che ci sono commit legati al coverage dei test

monen17 reviewed (changes requested):
Risolvere conflitti

eLBati commented:
/ocabot merge minor
